src/App.tsx

```diff
@@ -1,16 +1,43 @@
-import { useState } from 'react';
-import { Layout } from 'shared';
+import { useEffect, useState } from 'react';
+import { Route, Routes } from 'react-router-dom';
+import { Layout, NoMatch, ProtectedRoute } from 'shared';
+import { Blog, BlogDetails, Home, Info, Login } from 'views';
 
 export const App: React.FC = () => {
   const [isLoggedIn, setIsLogggedIn] = useState<boolean>(false);
 
+  useEffect(() => {
+    const storage = window.sessionStorage.getItem('loggedIn');
+    if (storage === null) {
+      window.sessionStorage.setItem('loggedIn', 'false');
+      return;
+    }
+    setIsLogggedIn(JSON.parse(storage));
+  }, []);
+
   function onLogin() {
     setIsLogggedIn(!isLoggedIn);
   }
 
+  useEffect(() => {
+    window.sessionStorage.setItem('loggedIn', `${isLoggedIn}`);
+  }, [onLogin]);
+
   return (
-    <Layout onLogin={() => onLogin()}>
-      <p>TODO: Add react router</p>
-    </Layout>
+    <Routes>
+      <Route path="/" element={<Layout onLogin={() => onLogin()} />}>
+        <Route index element={<Home />} />
+        <Route path="info" element={<Info />} />
+        <Route path="login" element={<Login isAuth={isLoggedIn} />} />
```

Comment:
Ovdje si mogao staviti npr.
element={isLoggedIn ? <Blog /> : <Navigate to="/login" replace />} pa si na taj nacin mogao izbjeci dodavanje logike u samu Login komponentu